rust/private/rustc.bzl

```diff
@@ -905,8 +905,11 @@ def construct_arguments(
     rustc_flags.add_all(rust_std_paths, before_each = "-L", format_each = "%s")
     rustc_flags.add_all(rust_flags)
 
+    # Gather data path from crate_info since it is inherited from real crate for rust_doc and rust_test
+    data_paths = getattr(attr, "data", []) + crate_info.compile_data_targets.to_list()
+
     # Deduplicate data paths due to https://github.com/bazelbuild/bazel/issues/14681
-    data_paths = depset(direct = getattr(attr, "data", []) + getattr(attr, "compile_data", [])).to_list()
+    data_paths = depset(direct = data_paths).to_list()
```

Comment:
Rather than flattening this list twice, can you instead initially construct a depset and then to_list it once?

Reply:
done